packages/vuetify/src/composables/filter.ts

```diff
@@ -0,0 +1,99 @@
+
+// Utilities
+import { getPropertyFromItem, propsFactory, wrapInArray } from '@/util'
+import { computed } from 'vue'
+
+// Types
+import type { PropType, Ref } from 'vue'
+
+export type FilterFunction = typeof defaultFilter
+export type FilterMode = 'intersection' | 'union'
+export type FilterKeys = string | string[]
+export type FilterKeysFn = Record<string, FilterFunction>
+
+export interface FilterProps {
+  filterFn?: FilterFunction
+  filterKeys?: FilterKeys
+  filterKeysFn?: FilterKeysFn
```

Comment:
Personally I feel like these are a bit confusing and will probably lead to a lot of repeated questions. The following are all hypothetical/devil's advocate questions, not me wanting answers.

What's the difference between `filterFn` and `filterKeysFn`?
Why is `filterFn` a function and `filterKeysFn` an object when they both end in `Fn`?
What's the relationship between `filterKeys` and `filterKeysFn`?

In this current setup, could you not combine filterKeys and filterKeysFn?

```
[
  'name',
  {
    key: 'title',
    filterFn: () => ...
  }
]
```